src/main/java/org/prebid/server/vast/VastModifier.java

```diff
@@ -86,14 +102,15 @@ private String appendTrackingUrlToVastXml(String vastXml, String vastUrlTracking
             return vastXml;
         }
 
-        final String impressionUrl = "<![CDATA[" + vastUrlTracking + "]]>";
-        final String openTag = "<Impression>";
+        final String impressionTag = "<Impression><![CDATA[" + vastUrlTracking + "]]></Impression>";
+        final String inlineCloseTag = IN_LINE_TAG.replace("<", "</");
 
-        // empty impression tag - just insert the link
-        if (closeTagIndex - vastXml.indexOf(openTag) == openTag.length()) {
-            return vastXml.replaceFirst(openTag, openTag + impressionUrl);
-        }
+        return vastXml.replace(inlineCloseTag, impressionTag + inlineCloseTag);
+    }
+
+    private String appendTrackingUrlForWrapperType(String vastXml, String vastUrlTracking, Integer wrapperTagIndex) {
+        final String impressionTag = "<Impression><![CDATA[" + vastUrlTracking + "]]></Impression>";
 
-        return vastXml.replaceFirst(closeTag, closeTag + openTag + impressionUrl + closeTag);
+        return vastXml.replaceFirst(WRAPPER_TAG, WRAPPER_TAG + impressionTag);
```

Comment:
It might be technically correct but still causes itching somehow. Can we insert impression tag in wrapper after the last present impression to be in line with how it is done in inline object?